src/components/VariantSearchField.tsx

```diff
@@ -25,10 +25,157 @@ const backgroundColor: { [key in SearchType]: string } = {
+  nsp16: 6799,
+};
+
+function getEquivalent(value: string, notation: string) {
+  let orf1aorbcodon: number | null = null;
+  let nsp: string | null = null;
+  let nspCodon: number | null = null;
+  let combinedCodon: number | null = 3646;
+  let mutationArray = value.split(':');
+  if (mutationArray[0] === 'ORF1b' || mutationArray[0] === 'ORF1a') {
+    let letterBefore = mutationArray[1].charAt(0);
+    let letterAfter = mutationArray[1].charAt(mutationArray[1].length - 1);
+    orf1aorbcodon = parseInt(mutationArray[1].slice(1, -1));
+    combinedCodon = orf1aorbcodon + (mutationArray[0] === 'ORF1b' ? 4401 : 0);
+    nsp = 'nsp1';
+    nspCodon = combinedCodon - nsps[nsp] + 1;
+    for (const [key, value] of Object.entries(nsps)) {
+      if (combinedCodon > value && combinedCodon - value < combinedCodon - nsps[nsp]) {
+        nsp = key;
+        nspCodon = combinedCodon - value + 1;
+      }
+    }
+    if (notation === 'nsp') {
+      return `${nsp}:${letterBefore}${nspCodon}${letterAfter}`;
+    } else if (notation === 'orf1ab') {
+      return `ORF1ab:${letterBefore}:${combinedCodon}${letterAfter}`;
+    } else {
+      return value;
+    }
+  }
+}
+
+const translateMutation = (oldValue: string) => {
+  let orf1aorb: string | null = null;
+  let orf1aorbcodon: number | null = null;
+  let nsp: string | null = null;
+
+  let nspCodon: number | null = null;
+  let combinedCodon: number | null = 3646;
+  let mutationArray = oldValue.split(':');
+
+  if (mutationArray[0] === 'ORF1b' || mutationArray[0] === 'ORF1a') {
+    let letterBefore = mutationArray[1].charAt(0);
+    let letterAfter = mutationArray[1].charAt(mutationArray[1].length - 1);
+    orf1aorbcodon = parseInt(mutationArray[1].slice(1, -1));
+    combinedCodon = orf1aorbcodon + (mutationArray[0] === 'ORF1b' ? 4401 : 0);
+    nsp = 'nsp1';
+    nspCodon = combinedCodon - nsps[nsp] + 1;
+    for (const [key, value] of Object.entries(nsps)) {
+      if (combinedCodon > value && combinedCodon - value < combinedCodon - nsps[nsp]) {
+        nsp = key;
+        nspCodon = combinedCodon - value + 1;
+      }
+    }
+
+    return `${nsp}:${letterBefore}${nspCodon}${letterAfter} = ORF1ab:${letterBefore}:${combinedCodon}${letterAfter}`;
+  } else if (mutationArray[0].toLowerCase().startsWith('nsp')) {
+    let letterAfter = !is_numeric(mutationArray[1].charAt(mutationArray[1].length - 1))
+      ? mutationArray[1].charAt(mutationArray[1].length - 1)
+      : '';
+
+    let letterBefore = !is_numeric(mutationArray[1].charAt(0)) ? mutationArray[1].charAt(0) : '';
+
+    if (letterAfter !== '') {
+      nspCodon = is_numeric(mutationArray[1].charAt(0))
+        ? parseInt(mutationArray[1].slice(0, -1))
+        : parseInt(mutationArray[1].slice(1, -1));
+    } else {
+      nspCodon = is_numeric(mutationArray[1].charAt(0))
+        ? parseInt(mutationArray[1])
+        : parseInt(mutationArray[1].slice(1));
+    }
+
+    nsp = mutationArray[0];
+    combinedCodon = nsps[nsp] + nspCodon - 1;
```

Comment:
The auto-suggestion should show nothing if the user types a non-existing NSP and should not translate it to `ORF1a:NaN`:

![image](https://user-images.githubusercontent.com/18666552/174026264-dd326283-57e0-40d3-8e1e-dc862b8b0ae2.png)